File: functions/documents/document-account.js
function DocumentAccount(doc)
{
	const data = doc.data()
	
	this.hasPassword = function(password)
	{
		return data.password == password
	}

	this.update = function(value)
	{
		return doc.ref.update(value)
	}

	this.gameRef = function(gameId)
	{
		return data.games.find(ref => ref.id == gameId)
	}

	this.json = function()
	{
		return {
			email: data.email,
			nickname: data.nickname,
			games: data.games.map(game => game.ref.id)
		}
	}
}

module.exports = DocumentAccount

File: src/main/kotlin/com/mauriciotogneri/momowars/documents/DocumentAccount.kt
package com.mauriciotogneri.momowars.documents

import com.mauriciotogneri.momowars.firebase.DocumentReference
import com.mauriciotogneri.momowars.firebase.DocumentSnapshot
import com.mauriciotogneri.momowars.utils.await
import kotlin.js.Json

class DocumentAccount(document: DocumentSnapshot)
{
    private val doc = document
    private val data = doc.data()

    val email: String = data.email
    val nickname: String = data.nickname
    val password: String = data.password
    val games: Array<DocumentReference> = data.games

    fun hasPassword(text: String): Boolean
    {
        return (password == text)
    }

    suspend fun update(value: Json)
    {
        doc.ref.update(value).await()
    }

    fun gameRef(gameId: String): DocumentReference?
    {
        return games.find { ref -> ref.id == gameId }
    }
}